library/src/main/java/bio/terra/landingzone/library/landingzones/definition/factories/CromwellStepsDefinitionProvider.java

```diff
@@ -61,82 +63,63 @@ public List<DefinitionVersion> availableVersions() {
   @Override
   public List<Pair<Step, RetryRule>> get(
       ArmManagers armManagers,
-      ParametersResolver parametersResolver,
+      ParametersResolverProvider parametersResolverProvider,
       ResourceNameProvider resourceNameProvider,
       LandingZoneProtectedDataConfiguration landingZoneProtectedDataConfiguration) {
     return List.of(
+        Pair.of(
```

Comment:
🤔 I am not sure what this comment means -- are saying get the MRG info and therefore the parametersresolver before the flight and pass it in as a first class arg rather than dynamically set in the working map?

Reply:
This was an old comment that I thought I'd deleted but that must've slipped through somehow. The reason for the comment originally was that `GetManagedResourceGroupInfo` used to happen after `ValidateLandingZoneParametersStep`. I had been trying to work out how to get a `ParametersResolver` for `ValidateLandingZoneParametersStep` when we didn't have the target MRG's region until `GetManagedResourceGroupInfo` and my solution was just to move that step earlier in the flight so we could create the `ParametersResolver` during the flight.